src/config/key.rs

```diff
@@ -47,6 +47,64 @@ pub fn parse_key(input: &str) -> Result<Key, Box<dyn Error>> {
         "SUPER_L" => Key::KEY_LEFTMETA,
         "WIN_R" => Key::KEY_RIGHTMETA,
         "WIN_L" => Key::KEY_LEFTMETA,
+
+        //Entirely custom scancodes
```

Comment:
Where did they come from? I'm not sure if I can maintain this portion. Is it possible to send this patch to evdev crate instead?

Reply:
Oh right, i only ever mentioned that change [here](https://github.com/k0kubun/xremap/issues/180#issuecomment-1288149131).

It's a way for Xremap to recognize names like "BTN_XRIGHTCURSOR" in config files as valid,
which is much simpler than modifying evdev,
and is also easier to modify.

In other words,
my code no longer uses a change in evdev's scancode enum,
but instead gives custom aliases to scancodes,
just like the code above said aliases.

An example for clarity :
`"WIN_L" => Key::KEY_LEFTMETA,`
is basically the same thing as
`"WIN_L" => evdev::Key(125),`
The first just uses the name KEY_LEFTMETA instead of the corresponding scancode.

Here's one of my custom aliases :
`"BTN_XRIGHTCURSOR" => evdev::Key(59974),`
There is no name corresponding to that scancode,
so my code has to use the scancode instead.


My code takes relative events,
"transforms" them into key events with those scancodes,
and then runs those key events through modmap and keymap.

To be able to map those key events to other events,
users need names to put in the config file,
which is where these aliases are used.

The reason the scancodes i use have no corresponding name is because no device emits them,
which is perfect,
since otherwise modmap/keymap wouldn't be able to tell the difference between those events and my disguised events.


I'll change my very undescriptive and misleading comment to better reflect what those aliases are and why they're used.



I'm not sure i see any possible problems maintaining these aliases,
since maintaining them and other aliases should be no different.

It should be possible to send the equivalent of this as a pull request to the evdev crate instead,
but they might not accept it,
and since i don't see why these aliases would be difficult to maintain,
I'm not sure that would be very useful.

Reply:
> Here's one of my custom aliases

Why do you need a custom alias?

> The reason the scancodes i use have no corresponding name is because no device emits them, which is perfect, since otherwise modmap/keymap wouldn't be able to tell the difference between those events and my disguised events.

Why do you need to disguise events?

Reply:
Perhaps you can add a config syntax to define those custom codes. I don't think xremap should be responsible for maintaining your original scancodes.

Reply:
(I'm assuming that you're still talking about custom aliases?)

>Perhaps you can add a config syntax to define those custom codes. 

I'm not sure i understand what you're proposing.

How would config syntaxes allow giving names to scancodes,
names usable in config files?
What's the difference between my custom aliases and the ones above?
As in, why is it better for config syntax to replace my custom aliases,
but not other custom aliases?

I can't think of a way to use config syntax that wouldn't lose the clarity these custom aliases can have.
For instance, which config file do you think is clearest :

The one using some config syntax to differentiate between positive and negative relative events?
```
modmap:
  - name: config syntax
    remap:
      positive: REL_X: KEY_H
      negative: REL_X: KEY_J
      
      positive: REL_WHEEL: KEY_E
      negative: REL_WHEEL: KEY_T
      
      positive: REL_Y: KEY_K
      negative: REL_Y: KEY_L
```

or the one using custom aliases to do the same thing?
```
modmap:
  - name: custom aliases
    remap:
      BTN_XUPCURSOR: KEY_H
      BTN_XDOWNCURSOR: KEY_J
      
      BTN_XDOWNSCROLL: KEY_E 
      BTN_XUPSCROLL: KEY_T
      
      BTN_XLEFTCURSOR: KEY_K
      BTN_XRIGHTCURSOR: KEY_L
```



>I don't think xremap should be responsible for maintaining your original scancodes.

You do maintain other custom aliases,
what would be different about mine?
If it's that you think the numbers¹ seem arbitrary and could cause confusion,
I could use the [offsetter const](https://github.com/GitPerseus/xremap/blob/feature/adds_relative_event_catcher/src/event_handler.rs#L139) to define them,
and even create a test to make sure the const isn't changed to a problematic number.


¹(the numbers used instead of names like in other custom aliases.)

Reply:
(I'm assuming that you're still talking about custom aliases?)

>Perhaps you can add a config syntax to define those custom codes. 

I'm not sure i understand what you're proposing.

How would config syntaxes allow giving names to scancodes,
names usable in config files?
What's the difference between my custom aliases and the ones above?
As in, why is it better for config syntax to replace my custom aliases,
but not other custom aliases?

I can't think of a way to use config syntax that wouldn't lose the clarity these custom aliases can have.
For instance, which config file do you think is clearest :

The one using some config syntax to differentiate between positive and negative relative events?
```
modmap:
  - name: config syntax
    remap:
      positive: REL_X: KEY_H
      negative: REL_X: KEY_J
      
      positive: REL_WHEEL: KEY_E
      negative: REL_WHEEL: KEY_T
      
      positive: REL_Y: KEY_K
      negative: REL_Y: KEY_L
```

or the one using custom aliases to do the same thing?
```
modmap:
  - name: custom aliases
    remap:
      BTN_XUPCURSOR: KEY_H
      BTN_XDOWNCURSOR: KEY_J
      
      BTN_XDOWNSCROLL: KEY_E 
      BTN_XUPSCROLL: KEY_T
      
      BTN_XLEFTCURSOR: KEY_K
      BTN_XRIGHTCURSOR: KEY_L
```



>I don't think xremap should be responsible for maintaining your original scancodes.

You do maintain other custom aliases,
what would be different about mine?
If it's that you think the numbers¹ seem arbitrary and could cause confusion,
I could use the [offsetter const](https://github.com/GitPerseus/xremap/blob/feature/adds_relative_event_catcher/src/event_handler.rs#L139) to define them,
and even create a test to make sure the const isn't changed to a problematic number.


¹(the numbers used instead of names like in other custom aliases.)

Reply:
> What's the difference between my custom aliases and the ones above?
> As in, why is it better for config syntax to replace my custom aliases, but not other custom aliases?

I think the point is, I believe existing custom aliases are something many people use, and I still don't understand why you would want custom aliases for your case.

Are they something not defined by evdev at all? Why can't we define them in evdev? The worst case scenario is that we define custom names first, the upstream evdev defines them later, and we end up needing to maintain both inconsistent names for backward compatibility. Are you sure it's not gonna happen?

Reply:
Also, can you consider addressing other comments I left? If the code is in a mergeable state, I might just merge it, but somehow you keep doing discussions. Reading code that is consistent with existing code and tests that verify what you want to do are sometimes more expressive than discussions.